<launch>
    <arg name="mav_name" default="firefly"/>
    <arg name="voxel_size" default="0.50"/>
    <arg name="num_agents" default="3"/>
    <arg name="param_file" default="$(find multi_robot_global_planner)/cfg/parameters_powerplant.yaml" />
    <arg name="waypoints_file" default="$(find multi_robot_global_planner)/cfg/waypoints_powerplant_intersection.yaml" />
    <arg name="launch_prefix" default=""/>
    <arg name="launch_prefix_voxblox" default=""/>

    <node name="multi_robot_global_planner" pkg="multi_robot_global_planner" type="multi_robot_global_planner_node" output="screen" args="-alsologtostderr" clear_params="true" launch-prefix="$(arg launch_prefix)">
       <rosparam command="load" file="$(arg param_file)" />
       <rosparam command="load" file="$(arg waypoints_file)" />
       <param name="voxel_size" value="$(arg voxel_size)"/>
       <param name="tsdf_voxel_size" value="$(arg voxel_size)"/>
       <remap from="multi_robot_global_planner/esdf_map_in" to="esdf_map" />
       <remap from="multi_robot_global_planner/tsdf_map_in" to="tsdf_map" />
    </node>

    <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true" launch-prefix="$(arg launch_prefix_voxblox)">

       <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
   	   <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />

       <remap from="pointcloud_transform0" to="/$(arg mav_name)_0/transform_pointcloud"/>
       <remap from="pointcloud_transform1" to="/$(arg mav_name)_1/transform_pointcloud"/>
       <remap from="pointcloud_transform2" to="/$(arg mav_name)_2/transform_pointcloud"/>

       <param name="use_tf_transforms" value="true"/>
       <param name="world_frame" value="world"/>
       <param name="num_agents" value="$(arg num_agents)"/>
       <remap from="transform" to="/pose_graph_backend/transform0"/>
       
       <param name="publish_tsdf_map" value="true" />
       <param name="publish_esdf_map" value="true" />
       <param name="publish_slices" value="true" />
       <param name="publish_pointclouds" value="true" />
       <param name="publish_traversable" value="false"/>

       <param name="use_freespace_pointcloud" value="false" />
       <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
       <param name="tsdf_voxels_per_side" value="16" />
       <param name="voxel_carving_enabled" value="true" />
       <param name="color_mode" value="normals" />
       <param name="verbose" value="false" />
       <param name="update_mesh_every_n_sec" value="1.0" />
       <param name="slice_level" value="-0.5" />
       <param name="method" value="fast" />
       <param name="use_const_weight" value="false" />
       <param name="max_ray_length_m" value="10"/>
       <param name="pointcloud_queue_size" value="20"/>
       <param name="min_ray_length_m" value="0.15"/>
    </node>

</launch>
